File: jagraj/24/singleNumber.py
from typing import List
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        counts = {}
        for i in nums:
            counts[i] = counts.get(i, 0)+1
        
        for k,v in counts.items():
            if v == 1:
                return k
        return -1

File: brady/single_number.java
class Solution {
  
    // O(n) time, O(1) space
    public int singleNumber(int[] nums) {
        int result = 0;
        for (int num : nums) result ^= num; // XOR with all of the nums, each pair of duplicates will cancel each other out
        return result;
    }
}
